### DISCUSSION TITLE
How to correctly log work for completed task?

### DISCUSSION THREAD

**Discussion by poetaman** [GothenburgBitFactory/taskwarrior#2513] [Q&A]
I have a task that I recently completed but forgot to change its status to start before starting to work on it. Given task was complete by the time I realized, I directly set the status to "complete". Then after reading the docs on ``task log``, I wonder if I could log work for the completed task? I tried following two commands in sequence:

1. ``task <uuid> start:21:00 rc.dateformat:hh:mm``
2. ``task <uuid> end:22:30 rc.dateformat:hh:mm`` 

But when I look at the information of task, it shows ``start deleted (duration:0:00:53)``. Does this mean it logged only 53 seconds? Shouldn't the duration be from last start to last end?

```
Thu 24 Jun 2021 22:30:15 Start set to 'Fri 25 Jun 2021 21:00:00'.
Thu 24 Jun 2021 22:30:41 End changed from 'Thu 24 Jun 2021 22:23:50' to 'Fri 25 Jun 2021 22:30:00'.
Thu 24 Jun 2021 22:31:08 Start deleted (duration: 0:00:53).
```

Additional corollary:

Perhaps there should be a way to specify task log time duration? Like ``task <uuid> log duration:1:30hrs``? That way if someone does not really remember what exact time they started and stopped working on a task, but do have a rough estimate, they can log duration they worked on it.

**Comment by tbabej** [comment]
I think the duration reporting in that output is a bug. Let's split corollary into a separate discussion :)

**Reply by tbabej** [reply]
OK, with regards to the duration itself, TW indeed computes the duration of the task in the `task info` output, but it does not accumulate total duration.

What you're proposing would be achievable with a User-Defined-Attribute for task duration.  Here's an example of a hook by @kostajh which uses a UDA for task duration and accumulates the total duration: https://github.com/kostajh/taskwarrior-time-tracking-hook

**Reply by tbabej** [reply]
The bug is now fixed on 2.6.0, see #2514 :slightly_smiling_face:
